### PR TITLE
Adding tests for animationStatus

### PR DESCRIPTION
**Pull request by goderbauer**
* Adds tests for AnimationController.status.
* Minor change to how AnimationStatus is reported for `animateTo` for consistency.
* Adds docs about the expected AnimationStatus for certain operations.

### CODE REVIEW

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 1]
```diff
@@ -270,6 +270,14 @@ class AnimationController extends Animation<double>
...
   /// canceled, meaning the future never completes and its [TickerFuture.orCancel]
   /// derivative future completes with a [TickerCanceled] error.
+  ///
+  /// During the animation, [status] is reported as [AnimationStatus.forward],
+  /// which switches to [AnimationStatus.completed] when [upperBound] is
+  /// reached at the end of the animation.
+  ///
+  /// You should not call [forward] or [reverse] on the same instance of an
+  /// [AnimationController] on which you also call [animateTo]. If you do mix
+  /// those calls, [status] might not have the expected value.
```
Comment (Hixie) [inline packages/flutter/lib/src/animation/animation_controller.dart]: i'd say "might not have the value you expect", rather than _the_ expected value.

Comment (Hixie) [inline packages/flutter/lib/src/animation/animation_controller.dart]: (if we say _the_ expected value, it leads to the question, why don't you fix it to give the expected value)

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 2]
```diff
@@ -294,6 +302,14 @@ class AnimationController extends Animation<double>
...
   /// canceled, meaning the future never completes and its [TickerFuture.orCancel]
   /// derivative future completes with a [TickerCanceled] error.
+  ///
+  /// During the animation, [status] is reported as [AnimationStatus.reverse],
+  /// which switches to [AnimationStatus.dismissed] when [lowerBound] is
+  /// reached at the end of the animation.
+  ///
+  /// You should not call [forward] or [reverse] on the same instance of an
+  /// [AnimationController] on which you also call [animateTo]. If you do mix
+  /// those calls, [status] might not have the expected value.
```
Comment (Hixie) [inline packages/flutter/lib/src/animation/animation_controller.dart]: ditto

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 3]
```diff
@@ -318,7 +334,21 @@ class AnimationController extends Animation<double>
...
   /// derivative future completes with a [TickerCanceled] error.
+  ///
+  /// During the animation, [status] is reported as [AnimationStatus.forward]
+  /// regardless of whether `target` > [value] or not. At the end of the
+  /// animation, when `target` is reached, [status] is reported as
+  /// [AnimationStatus.completed].
+  ///
+  /// You should not call [forward] or [reverse] on the same instance of an
+  /// [AnimationController] on which you also call [animateTo]. If you do mix
+  /// those calls, [status] might not have the expected value.
```
Comment (Hixie) [inline packages/flutter/lib/src/animation/animation_controller.dart]: ditto

Comment (Hixie) [inline packages/flutter/lib/src/animation/animation_controller.dart]: Except that now that you've made this change, why won't it have the right value? You've removed the confusion, no?

This seems like a great change, if none of the tests fail.

File: packages/flutter/test/animation/animation_controller_test.dart [hunk 4]
```diff
@@ -395,4 +395,78 @@ void main() {
...
+    expect(statusLog, equals(<AnimationStatus>[ AnimationStatus.forward, AnimationStatus.completed ]));
+    statusLog.clear();
+
+    // Animate from 0.5 to 1.0
+    controller.animateTo(0.0);
+    tick(const Duration(milliseconds: 0));
+    tick(const Duration(milliseconds: 150));
+    expect(controller.value, 0.0);
+    expect(statusLog, equals(<AnimationStatus>[ AnimationStatus.forward, AnimationStatus.completed ]));
+    statusLog.clear();
```
Comment (Hixie) [inline packages/flutter/test/animation/animation_controller_test.dart]: you should test that forward/reverse don't affect the results

File: packages/flutter/lib/src/animation/animation_controller.dart (hunk 3 above)
Comment (goderbauer) [inline packages/flutter/lib/src/animation/animation_controller.dart]: I removed it.

File: packages/flutter/test/animation/animation_controller_test.dart (hunk 4 above)
Comment (goderbauer) [inline packages/flutter/test/animation/animation_controller_test.dart]: Added more tests for that.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**